operator/src/main/java/oracle/kubernetes/operator/helpers/DomainHomeHelper.java

```diff
@@ -0,0 +1,200 @@
+import oracle.kubernetes.operator.LabelConstants;
+import oracle.kubernetes.operator.logging.LoggingFacade;
+import oracle.kubernetes.operator.logging.LoggingFactory;
+import oracle.kubernetes.operator.work.Container;
+import oracle.kubernetes.operator.work.ContainerResolver;
+import oracle.kubernetes.operator.work.NextAction;
+import oracle.kubernetes.operator.work.Packet;
+import oracle.kubernetes.operator.work.Step;
+import oracle.kubernetes.weblogic.domain.v1.Domain;
+import oracle.kubernetes.weblogic.domain.v1.DomainSpec;
+
+public class DomainHomeHelper {
+  private static final LoggingFacade LOGGER = LoggingFactory.getLogger("Operator", "Operator");
+
+  private DomainHomeHelper() {}
+  
+  public static Step createDomainHomeStep(Step next) {
+    return new DomainHomeStep(next);
+  }
+  
+  public static class DomainHomeStep extends Step {
+    public DomainHomeStep(Step next) {
+      super(next);
+    }
+
+    @Override
+    public NextAction apply(Packet packet) {
+      Container c = ContainerResolver.getInstance().getContainer();
+      CallBuilderFactory factory = c.getSPI(CallBuilderFactory.class);
+
+      DomainPresenceInfo info = packet.getSPI(DomainPresenceInfo.class);
+      Domain dom = info.getDomain();
+      V1ObjectMeta meta = dom.getMetadata();
+      String namespace = meta.getNamespace();
+
+      V1Job job = computeDomainHomeJob(info);
+      Step conflictStep = factory.create().deleteJobAsync(job.getMetadata().getName(), namespace, new ResponseStep<V1Status>(next) {
+        @Override
+        public NextAction onFailure(Packet packet, ApiException e, int statusCode,
+            Map<String, List<String>> responseHeaders) {
+          if (statusCode == CallBuilder.NOT_FOUND) {
+            return onSuccess(packet, null, statusCode, responseHeaders);
+          }
+          return super.onFailure(packet, e, statusCode, responseHeaders);
+        }
+
+        @Override
+        public NextAction onSuccess(Packet packet, V1Status result, int statusCode,
+            Map<String, List<String>> responseHeaders) {
+          return doNext(DomainHomeStep.this, packet);
+        }
+      });
+      
+      Step create = factory.create().createJobAsync(namespace, computeDomainHomeJob(info), new ResponseStep<V1Job>(next) {
+        @Override
+        public NextAction onFailure(Packet packet, ApiException e, int statusCode,
+            Map<String, List<String>> responseHeaders) {
+          return super.onFailure(conflictStep, packet, e, statusCode, responseHeaders);
+        }
+        
+        @Override
+        public NextAction onSuccess(Packet packet, V1Job result, int statusCode,
+            Map<String, List<String>> responseHeaders) {
+          // TODO: wait for the job to finish or fail
+          //       update domain status condition
+          return null;
+        }
+      });
+      
+      return doNext(create, packet);
+    }
+    
+    protected V1Job computeDomainHomeJob(DomainPresenceInfo info) {
```

Comment:
Need a unit test for this method.

Reply:
Currently, create-weblogic-domain.sh makes a copy of create-weblogic-domain-job-template.yaml and customizes it, then loads it into kubernetes to create the job.

It looks you're changing this - i.e. having code in the operator directly create the job.

But it looks like you've only partially done the work - I don't see changes to create-weblogic-domain.sh.

Perhaps this code snuck in by mistake from the WDT branch?

(p.s. it's also missing javadoc)